src/components/range-inputs.tsx

```diff
@@ -0,0 +1,303 @@
+                    clickCallback={handleTimeClear}
+                    show
+                />
+            </div>
+        </div>
+    );
+};
+
+type RangeInputProps = {
+    placeholder?: string,
+    classes?: string,
+    reference: React.RefObject<HTMLInputElement>,
+    timeRef: React.RefObject<HTMLInputElement>,
+    type: string,
+    id: string,
+    value: string,
+    handleChange: ((value: string) => void)
+};
+
+const RangeInput = ({ placeholder = 'Enter', classes = '', reference, timeRef, type, value, id, handleChange }: RangeInputProps): JSX.Element => {
+
+    const changeHandler = () => handleChange(reference?.current?.value || '');
+
+    const clearInput = () => {
+        if (reference?.current) reference.current.value = '';
+        changeHandler();
+    }
+
+    return (
+        type === 'timestamp' ? (
+            <DateTimePicker classes={classes} id={id} value={value} dateRef={reference} timeRef={timeRef} handleChange={handleChange} />
+        ) : (
+            <div className='chaise-input-control has-feedback'>
+                {
+                    type === 'int' || type === 'float' ? <input id={id} type='number' placeholder={placeholder} className={classes}
+                        ref={reference} onChange={changeHandler} />
+                        : <input id={id} type='date' className={classes} ref={reference} step='1'
+                            defaultValue={value} pattern='\d{4}-\d{2}-\d{2}' min='1970-01-01' max='2999-12-31' onChange={changeHandler} />
+                }
+                <ClearInputBtn
+                    btnClassName='range-input-clear'
+                    clickCallback={clearInput}
+                    show
+                />
+            </div>
+        )
+
+    )
+};
+
+
+type RangeInputHOCProps = {
+    /**
+     * an string for the type of input expected:
+     * int
+     * float
+     * date
+     * timestamp
+     */
+    inputType: string,
+    /**
+     * use for applying custom styles on the component
+     */
+    classes?: string
+};
+
+const getType = (inputType: string): string => {
+
+    if (inputType.indexOf('int') > -1) return 'int';
+
+    if (inputType.indexOf('float') > -1) return 'float';
+
+    if (inputType.indexOf('date') > -1) return 'date';
+
+    if (inputType.indexOf('timestamp') > -1) return 'timestamp';
+
+    return 'Invalid_Type';
+}
+
+const RangeInputHOC = ({ inputType, classes }: RangeInputHOCProps) => {
+    const momentJS = windowRef.moment;
+
+    const fromRef = useRef<HTMLInputElement>(null);
+
+    const toRef = useRef<HTMLInputElement>(null);
+
+    const fromTimeRef = useRef<HTMLInputElement>(null);
+
+    const toTimeRef = useRef<HTMLInputElement>(null);
+
+    const type = getType(inputType);
+
+    const [error, setError] = useState<string | null>(null);
+
+    const [disableSubmit, setDisableSubmit] = useState<boolean>(type === 'int' || type === 'float');
+
+    const disableSubmitBtn = () => {
```

Comment:
Is it possible to combine disable/enable into one function call?

something like:
```
toggleSubmitBtn(condition) => {
  setDisableSubmit(condition)
}
```